### ISSUE TITLE
 java.io.IOException: Resetting to invalid mark

### ISSUE THREAD

**Issue by w2hhda** [BigBadaboom/androidsvg#181]
SVG.getFromInputStream(new FileInputStream(xxx)) cause java.io.IOException: Resetting to invalid mark

![image](https://user-images.githubusercontent.com/5924606/63834997-2346a880-c9a9-11e9-80fa-78a54674ce8d.png)

svg in flower_pot.zip

[flower_pot.zip](https://github.com/BigBadaboom/androidsvg/files/3549599/flower_pot.zip)

**Comment by BigBadaboom** [comment]
Hi @w2hhda, thanks for the report.

This was a bit tricky to find the cause of.  But I have worked out a fix.

**Comment by BigBadaboom** [comment]
Fixed.

Use a cleaner method to check for entities, reset the input stream, and switch to the SAX parser.
The problem with the previous method was that it appears the XmlPullParser uses reset() itself while parsing.  This was preventing us resetting it ourselves.  I'm pretty sure XPP wasn't doing that when I first wrote this code. Perhaps Android switched to a different implementation of XmlPullParser recently.
